app/src/main/java/com/github/se/project/model/notification/PushNotificationsService.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.se.project.model.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import com.github.se.project.R
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+class PushNotificationsService : FirebaseMessagingService() {
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    super.onMessageReceived(remoteMessage)
+    // Log.d("FCM", "Message received: ${remoteMessage.data}")
+
+    // Show notification
+    val title = remoteMessage.notification?.title ?: "Default Title"
+    val message = remoteMessage.notification?.body ?: "Default Message"
+    showNotification(title, message)
+  }
+
+  private fun showNotification(title: String, message: String) {
+    val channelId = "default_channel_id"
+    val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      val channel =
+          NotificationChannel(channelId, "Default Channel", NotificationManager.IMPORTANCE_DEFAULT)
+      notificationManager.createNotificationChannel(channel)
+    }
+
+    val notification =
+        NotificationCompat.Builder(this, channelId)
+            .setSmallIcon(R.drawable.logopocket)
+            .setContentTitle(title)
+            .setContentText(message)
+            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+            .build()
+
+    notificationManager.notify(0, notification)
```

Comment:
static Magic number is probably not the ideal implementation, especially if multiple notificatoin need to co exist, it might be better use the time of the notification as id for example